src/Connection.h

```diff
@@ -70,7 +70,7 @@ class Connection : public QObject {
 		/// qmCrypt locks access to csCrypt.
 		QMutex qmCrypt;
 #endif
-		CryptState csCrypt;
```

Comment:
First off I'd prefer this to remain a non-pointer type.
I guess however that c++ won't allow this, right? (not sure atm)

In that case you should at least use a smart-pointer (probably a unique pointer as we don't want to share ownership of this member).

Reply:
I tried to initialize a `CryptState csCrypt`, with `csCrypt = CryptStateOCB2();` and it would give me `error: variable type 'CryptState' is an abstract class`. I think pointer is the only solution.

Reply:
Yeah it probably has to do with the fact that the compiler needs to know the size of the member variable which is unknown if it could be any of the sub-classes...